cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/AbstractReactiveLimitsOwnerConversion.java

```diff
@@ -32,49 +36,89 @@ protected void convertReactiveLimits(Generator g) {
         // ReactiveCapabilityCurve.
         if (p.containsKey("ReactiveCapabilityCurve")) {
             String curveId = p.getId("ReactiveCapabilityCurve");
-            ReactiveCapabilityCurveAdder rcca = g.newReactiveCapabilityCurve();
             PropertyBags curveData = context.reactiveCapabilityCurveData(curveId);
-            curveData.forEach(d -> {
+            if (curveData.size() >= 2) {
+
+                Map<Double, Range<Double>> qRanges = new HashMap<>();
+                curveData.forEach(d -> {
+                    double p = d.asDouble("xvalue");
+                    double minQ = d.asDouble("y1value");
+                    double maxQ = d.asDouble("y2value");
+
+                    if (!checkPointValidity(p, minQ, maxQ)) {
+                        return;
+                    }
+
+                    Range<Double> qRange = fixedMinMaxQ("ReactiveCapabilityCurve Point", minQ, maxQ);
+
+                    // check if there is a previous point with same p
+                    Range<Double> prev = qRanges.get(p);
+                    if (prev != null) {
+                        if (prev.isConnected(qRange)) {
+                            fixed("reactive capability curve", String.format("point merged with another one with same p (%f)", p));
+                            qRanges.put(p, prev.span(qRange));
+                        } else {
+                            ignored("reactive capability curve point", String.format("another point with same p (%f) and a disconnected reactive range", p));
+                        }
+                    } else {
+                        qRanges.put(p, qRange);
+                    }
+                });
+
+                // create curve
+                ReactiveCapabilityCurveAdder rcca = g.newReactiveCapabilityCurve();
+                for (Map.Entry<Double, Range<Double>> e : qRanges.entrySet()) {
```

Comment:
What happens here if after checking duplicated points, `qRange` size is 1?

Reply:
Fixed in the new algo, check on point count is done after removing duplicate points.